### PR TITLE
Decouple time interface from libultra

### PR DESCRIPTION
**Pull request by GiacomoGarbin**
PR for issue #65.

### CODE REVIEW

File: src/savefile/savefile.c [hunk 1]
```diff
@@ -26,13 +25,11 @@ extern OSMesgQueue dmaMessageQ;
 #define SRAM_pageSize    0xd 
 #define SRAM_relDuration 0x2
 
-#define SRAM_CHUNK_DELAY        OS_USEC_TO_CYCLES(10 * 1000)
+#define SRAM_CHUNK_DELAY (10 * 1000)
```
Comment (mwpenny) [inline src/savefile/savefile.c:28]: Minor: can you include the units in the name (i.e., `SRAM_CHUNK_DELAY_USECS`)? Makes it unambiguous to readers.

File: src/savefile/savefile.c [hunk 2]
```diff
@@ -125,7 +118,7 @@ void savefileLoad() {
     gSramHandle.domain = PI_DOMAIN2;
     gSramHandle.speed = 0;
 
-    osCreateMesgQueue(&timerQueue, &timerQueueBuf, 1);
+    timeInit();
```
Comment (mwpenny) [inline src/savefile/savefile.c:121]: Sleeping is only used here, but the time system as a whole is unrelated to the savefile. Can you move initialization to `main()`?

**Review by mwpenny — CHANGES_REQUESTED** [review]
Regarding your question in your first commit message:
> TODO: should we move time.h and time_libultra.c to the src/system folder?

Yes, let's follow the same pattern as we did for the controller.

**Comment by GiacomoGarbin** [comment]
> Yes, let's follow the same pattern as we did for the controller.

Done in [0f6052a](https://github.com/mwpenny/portal64-still-alive/pull/66/commits/0f6052a9ae46c2acb607b5c6d7b3defb5a993bbd).

Thanks for the review! I had also considered the points you suggested.

**Comment by mwpenny** [comment]
Great, thanks!

**Pull request merged**